src/app/components/search-results/search-results.component.ts

```diff
@@ -2,103 +2,115 @@ import { Component, OnInit } from '@angular/core';
 import { ActivatedRoute, Router } from '@angular/router';
 import { PageEvent } from '@angular/material/paginator';
 import { CkanService } from '../../services/ckan.service';
+import { Filter } from 'src/app/interfaces/filter';
+import { PartialDataset } from 'src/app/interfaces/dataset';
 
 @Component({
   selector: 'app-search-results',
   templateUrl: './search-results.component.html',
-  styleUrls: ['./search-results.component.scss']
+  styleUrls: ['./search-results.component.scss'],
 })
 export class SearchResultsComponent implements OnInit {
-  allResults: any[] = [];
-  filteredResults: any[] = [];
-
-  uniquePublishers: string[] = [];
-  selectedPublishers: string[] = [];
-  filteredPublishers: string[] = [];
+  allResults: PartialDataset[] = [];
+  results: PartialDataset[] = [];
+  totalResults: number = 0;
 
-  searchTerms = {
-    publisher: '',
-    // ... other search terms for different filters
-  };
+  currentSearchQuery: string = '';
+  currentFilters: Filter[] = [];
+  currentFilterQuery: string = '';
 
-  totalResults: number = 0;
   pageSize: number = 12;
   currentPage: number = 0;
   pageSizeOptions: number[] = [12, 24, 48];
-  currentSearchQuery: string = '';
 
-  constructor(private route: ActivatedRoute, private ckanService: CkanService, private router: Router) { }
+  constructor(
+    private route: ActivatedRoute,
+    private ckanService: CkanService,
+    private router: Router
+  ) {}
 
   ngOnInit(): void {
-    this.route.queryParams.subscribe(params => {
-      const query = params['query'];
-      this.loadSearchResults(query);
+    // Retrieve all values for filters (must go through all the pages)
+    this.ckanService.searchDatasets('', '', 0, 1_000_000_000_000).subscribe((data) => {
```

Comment:
I wonder if this is correct. There should be better ways to retrieve filter values without the need of querying datasets.